compiler/rustc_const_eval/src/interpret/memory.rs

```diff
@@ -953,7 +953,12 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
 
         // # Global allocations
         if let Some(global_alloc) = self.tcx.try_get_global_alloc(id) {
-            let (size, align) = global_alloc.size_and_align(*self.tcx, self.typing_env);
+            let (size, mut align) = global_alloc.size_and_align(*self.tcx, self.typing_env);
+            if let GlobalAlloc::Static(def_id) = global_alloc {
+                if let Some(static_align) = self.tcx.codegen_fn_attrs(def_id).alignment {
+                    align = Ord::max(align, static_align);
+                }
+            }
```

Comment:
So every other user of `global_alloc.size_and_align` still gets the wrong answer?